runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtPlugin.java

```diff
@@ -0,0 +1,478 @@
+)
+@Slf4j
+public class WintertodtPlugin extends Plugin
+{
+	static final int WINTERTODT_REGION = 6462;
+
+	@Inject
+	private Notifier notifier;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private OverlayManager overlayManager;
+
+	@Inject
+	private WintertodtOverlay overlay;
+
+	@Inject
+	private WintertodtConfig config;
+
+	@Inject
+	private ChatMessageManager chatMessageManager;
+
+	@Getter(AccessLevel.PACKAGE)
+	private WintertodtActivity currentActivity = WintertodtActivity.IDLE;
+
+	@Getter(AccessLevel.PACKAGE)
+	private int inventoryScore;
+
+	@Getter(AccessLevel.PACKAGE)
+	private int totalPotentialinventoryScore;
+
+	@Getter(AccessLevel.PACKAGE)
+	private int numLogs;
+
+	@Getter(AccessLevel.PACKAGE)
+	private int numKindling;
+
+	private Instant lastActionTime;
+
+	@Getter(AccessLevel.PACKAGE)
+	private boolean isInWintertodt;
+
+	@Provides
+	WintertodtConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(WintertodtConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		reset();
+		overlayManager.add(overlay);
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		overlayManager.remove(overlay);
+		reset();
+	}
+
+	private void reset()
+	{
+		inventoryScore = 0;
+		totalPotentialinventoryScore = 0;
+		numLogs = 0;
+		numKindling = 0;
+		currentActivity = WintertodtActivity.IDLE;
+	}
+
+	private boolean isInWintertodtRegion()
+	{
+		if (client.getLocalPlayer() != null)
+		{
+			return client.getLocalPlayer().getWorldLocation().getRegionID() == WINTERTODT_REGION;
+		}
+
+		return false;
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick gameTick)
+	{
+		if (!isInWintertodtRegion())
+		{
+			if (isInWintertodt)
+			{
+				log.debug("Left Wintertodt!");
+				reset();
+			}
+
+			isInWintertodt = false;
+			return;
+		}
+
+		if (!isInWintertodt)
+		{
+			reset();
+			log.debug("Entered Wintertodt!");
+		}
+		isInWintertodt = true;
+
+		checkActionTimeout();
+	}
+
+	private void checkActionTimeout()
+	{
+		if (currentActivity == WintertodtActivity.IDLE)
+			return;
+
+		int currentAnimation = client.getLocalPlayer() != null ? client.getLocalPlayer().getAnimation() : -1;
+		if (currentAnimation != IDLE)
+		{
+			return;
+		}
+
+		Duration actionTimeout = Duration.ofSeconds(3);
+		Duration sinceAction = Duration.between(lastActionTime, Instant.now());
+
+		if (sinceAction.compareTo(actionTimeout) >= 0)
+		{
+			log.debug("Activity timeout!");
+			currentActivity = WintertodtActivity.IDLE;
+		}
+	}
+
+	@Subscribe
+	public void onSetMessage(SetMessage setMessage)
+	{
+		if (!isInWintertodt)
+		{
+			return;
+		}
+
+		ChatMessageType chatMessageType = setMessage.getType();
+
+		if (chatMessageType != ChatMessageType.SERVER && chatMessageType != ChatMessageType.FILTERED)
+		{
+			return;
+		}
+
+		MessageNode messageNode = setMessage.getMessageNode();
+		WintertodtInterruptType interruptType = null;
+
+		if (messageNode.getValue().startsWith("The cold of"))
```

Comment:
Maybe this bunch of ifs could be part of the enum? Like, WintertodtInterruptType.fromMessage()